src/wms/src/main/java/org/geoserver/wms/featureinfo/LayerIdentifier.java

```diff
@@ -35,4 +35,12 @@
      */
     List<FeatureCollection> identify(FeatureInfoRequestParameters params, int maxFeatures)
             throws Exception;
+
```

Comment:
This method is actually used only inside the implementations, but never called from outside. So, no need to make it public in the interface, nothing holding a LayerIdentifer uses it. As far as I can see, you can remove it from here, remove the @Override markers from the implementations, and it will still work fine. Compare with the identify method, which sees usage from objects holding LayerIdentifier instead.

Reply:
will do